components/transaction/UserGemsButton.tsx

```diff
@@ -0,0 +1,47 @@
+import {StyleSheet, View, ActivityIndicator} from 'react-native';
+import Colors from '../../theme/colors';
+import { useGetTransactionQuery } from './api';
+import FetchError from '../common/feedback/FetchError';
+import CenteredFeedback from '../common/feedback/CenteredFeedback';
+import React from "react";
+import GemsIndicatorButton from "./GemsIndicatorButton";
+
+const UserGems = () => {
+  const { data, isError, isLoading } = useGetTransactionQuery();
+
+  if (isLoading) {
+    return (
+      <View style={styles.root}>
+        <ActivityIndicator size={40} color={Colors.gray[0]} />
+      </View>
+    );
+  }
+
+  if (isError) {
+    return <FetchError />;
+  }
+
+  if (!data) {
+    return <CenteredFeedback message='Cannot access gems info.' />;
+  }
+
+  return (
+    <GemsIndicatorButton gemCount={data.gems} handleOpenModal={() => {}} />
+  );
+}
+
+const styles = StyleSheet.create({
+  root: {
+    marginLeft: 120,
```

Comment:
You can use:
display: flex,
width: 100%,
justify-content: center,
align-items: center,